undertow/src/main/java/org/wildfly/extension/undertow/deployment/SharedSessionManagerDeploymentProcessor.java

```diff
@@ -114,7 +115,12 @@ public Duration getDefaultSessionTimeout() {
             }
         };
         provider.getSessionManagerFactoryServiceConfigurator(managerServiceName, configuration).configure(support).build(target).install();
+        // Skip route locator service installation in non-distributable case
+        if (provider.getRouteLocatorServiceConfigurator(configuration) != null) {
+            provider.getRouteLocatorServiceConfigurator(configuration).configure(support).build(target).install();
```

Comment:
Why create this CapaabilityServiceConfigurator twice?

Reply:
Fixed. Perhaps these would better be called `createXYZServiceConfigurator` instead of get `XYZServiceConfigurator`.